EventFilter/RawDataCollector/src/RawDataMapperByLabel.cc

```diff
@@ -0,0 +1,66 @@
+/** \file
+ * Implementation of class RawDataMapperByLabel
+ *
+ */
+
+#include "EventFilter/RawDataCollector/interface/RawDataMapperByLabel.h"
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h" 
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "DataFormats/Provenance/interface/ProcessHistory.h" 
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <iostream>
+
+using namespace edm;
+
+RawDataMapperByLabel::RawDataMapperByLabel(const edm::ParameterSet& pset) {
+
+  inputTags_ = pset.getParameter<std::vector<InputTag> >("RawCollectionList");
+  mainCollectionTag_ = pset.getParameter<InputTag>("MainCollection");
+  
+
+  inputTokens_.reserve(inputTags_.size());
+  for(tag_iterator_t inputTag = inputTags_.begin(); inputTag != inputTags_.end(); ++inputTag ) {
+    inputTokens_.push_back(consumes<FEDRawDataCollection>(*inputTag));
+  }
+  produces<FEDRawDataCollection>();
+  firstEvent_= true;
+  filledCollectionName_= InputTag("");
+}
+
+RawDataMapperByLabel::~RawDataMapperByLabel(){
+
+}
+
+
+void RawDataMapperByLabel::produce(Event & e, const EventSetup& c){
+
+ 
+ bool AlreadyACollectionFilled= false;
+ tag_iterator_t inputTag = inputTags_.begin();
+ for(tok_iterator_t inputTok = inputTokens_.begin(); inputTok != inputTokens_.end(); ++inputTok, ++inputTag  ) {
+   Handle<FEDRawDataCollection> input;
+   if (e.getByToken(*inputTok,input)){
+    	if(input.isValid()){
+    	    if(firstEvent_){ 
+    	        if(mainCollectionTag_==*inputTag) continue;
```

Comment:
@icali could you please address this comment?

Reply:
The only way that I have to don't include a double loop is to remove the condition: if(mainCollectionTag_==*inputTag) continue;
and add:
if(!(mainCollectionTag_==filledCollectionName_)) e.put(std::make_unique<FEDRawDataCollection>(*input.product()));

Not sure however if this feature makes sense as it would be enough to don't include the rawDataCollector original collection in the input of this module.
Please advise.